cli/http_util.rs

```diff
@@ -246,77 +231,204 @@ impl HttpClient {
         ..Default::default()
       },
       root_cert_store_provider,
-      cell: Default::default(),
+      clients_by_thread_id: Default::default(),
     }
   }
 
-  #[cfg(test)]
-  pub fn from_client(client: reqwest::Client) -> Self {
-    let result = Self {
-      options: Default::default(),
-      root_cert_store_provider: Default::default(),
-      cell: Default::default(),
-    };
-    result.cell.set(client).unwrap();
-    result
+  // todo(dsherret): improve/revamp `HttpClient`'s api in order to
+  // remove this and make `ReqwestClient`
```

Comment:
Do you mean `ReqwestClient` or `reqwest::Client`? If the latter then I think there's no way around it if we want to disallow sending across threads

Reply:
Oh, this is an old todo with a previous experimental API I had. I'll remove this.